packages/server/socketHelpers/sendEncodedMessage.ts

```diff
@@ -1,16 +1,56 @@
 import {HttpResponse, WebSocket} from 'uWebSockets.js'
+import handleClose from '../socketHandlers/handleClose'
 import sendSSEMessage from '../sse/sendSSEMessage'
+import ConnectionContext from './ConnectionContext'
 import isHttpResponse from './isHttpResponse'
 
 const ESTIMATED_MTU = 1400
-const sendEncodedMessage = (transport: WebSocket | HttpResponse, message: object | string) => {
-  if (transport.done) return
-  const str = JSON.stringify(message)
-  if (isHttpResponse(transport)) {
-    sendSSEMessage(transport as HttpResponse, str)
+const TIME_OUT_COEFFICIENT = 2.0
+const MAX_TIMEOUT = 10000 // 10 seconds
+const MAX_MESSAGE_ID = 128
+
+const sendAndPushToReliableQueue = (
+  context: ConnectionContext,
+  synId: number,
+  message: string,
+  timeout = 2000
+) => {
+  const {socket, reliableQueue} = context
+  if (timeout > MAX_TIMEOUT) {
+    isHttpResponse(socket) ? socket.close() : handleClose(socket)
     return
   }
-  transport.send(str, false, str.length > ESTIMATED_MTU)
+  sendEncodedMessageBasedOnSocket(socket, message)
+  const timer = setTimeout(() => {
+    const newTimeout = timeout * TIME_OUT_COEFFICIENT
+    sendAndPushToReliableQueue(context, synId, message, newTimeout)
```

Comment:
if this message failed to ack twice, this synId could have been created up to 8 seconds ago!
is it possible that a client gets 128 messages within 8 seconds? you bet!
instead of reusing an 8-second old synId, we should get a new one.
in other words, this function should call synId = context.synId++.

I'd suggest adding an API to the connectionContext like this:

```ts
getSynId() {
  this.synId++
  if (this.synId > MAX_ID) this.synId = 0;
  return this.synId
}
```
then, this function can just call context.getSynId()